gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/AbstractReplicationData.java

```diff
@@ -0,0 +1,55 @@
+package gobblin.data.management.copy.replication;
+
+import gobblin.data.management.copy.CopyableDataset;
+import gobblin.dataset.DatasetsFinder;
+
+/**
+ * Abstract class which has the basic implementations 
+ * @author mitu
+ *
+ */
+
+public abstract class AbstractReplicationData implements ReplicationData<CopyableDataset> {
+
+  protected final String replicationName;
+  protected final ReplicationLocation location;
+  protected final DatasetsFinder<CopyableDataset> datasetFinder;
+
+  protected AbstractReplicationData(String replicationName, ReplicationLocation location, DatasetsFinder<CopyableDataset> datasetFinder ) {
+    this.replicationName = replicationName;
+    this.location = location;
+    this.datasetFinder = datasetFinder;
+  }
+
+  @Override
+  public String getReplicationName() {
+    return this.replicationName;
+  }
+
+  @Override
+  public ReplicationLocation getReplicationLocation() {
+    return this.location;
+  }
+  
+  @Override
+  public DatasetsFinder<CopyableDataset> getDatasetsFinder(){
+    return this.datasetFinder;
+  }
+
+  @Override
+  public String toString() {
```

Comment:
Have a look at Objects.toStringHelper . It can help a lot with the implementation of methods like this one.